client/core/trade.go

```diff
@@ -1239,14 +1269,107 @@ func (t *trackedTrade) isSwappable(ctx context.Context, match *matchTracker) (re
 				"but we have not seen the counter-party's redemption yet. This could just"+
 				" be network latency.", match)
 		}
+
 		match.setSwapConfirms(int64(confs))
+
 		t.notify(newMatchNote(TopicConfirms, "", "", db.Data, t, match))
 		return false, false
 	}
 
 	return false, false
 }
 
+// updateDynamicSwapOrRedemptionFeesPaid updates the fees used for dynamic fee
+// checker transactions. We do not know the exact fees the tx will use until
+// they are mined, so this waits until they are mined and updates the value for
+// the entire trade.
+//
+// TODO: This method does not persist if the client is shut down. It probably
+// should because payed fees will be incorrect forever for this trade if not.
+func (t *trackedTrade) updateDynamicSwapOrRedemptionFeesPaid(ctx context.Context, match *matchTracker, isInit bool) {
+	wallet := t.wallets.fromWallet
+	if !isInit {
+		wallet = t.wallets.toWallet
+	}
+	feeChecker, dynamic := wallet.Wallet.(asset.DynamicSwapOrRedemptionFeeChecker)
+	if !dynamic {
+		return
+	}
+	txType := "swap"
+	if !isInit {
+		txType = "redemption"
+	}
+	var coinID []byte
+	// Check if a swap or redeem coin id has been populated in the
+	// match tracker. If it has we ask the wallet for the fees paid
+	// and add that to either the total swap or redeem fees for the
+	// trade.
+	check := func() (feeIsConfirmed bool) {
+		if isInit {
+			coinID = []byte(match.MetaData.Proof.MakerSwap)
+			if match.Side != order.Maker {
+				coinID = []byte(match.MetaData.Proof.TakerSwap)
+			}
+		} else {
+			coinID = []byte(match.MetaData.Proof.MakerRedeem)
+			if match.Side != order.Maker {
+				coinID = []byte(match.MetaData.Proof.TakerRedeem)
+			}
+		}
+		contractData := match.MetaData.Proof.ContractData
+		secretHash := match.MetaData.Proof.SecretHash
+		if coinID == nil {
+			return false
+		}
+		confs, err := wallet.RegFeeConfirmations(ctx, coinID)
+		if err != nil {
+			if errors.Is(err, asset.CoinNotFoundError) {
+				return false
+			}
+			t.dc.log.Errorf("Failed to determine actual %s transaction fees paid for "+
+				"match %s: %v", txType, match, err)
+			return true
+		}
+		if confs < updateDynamicSwapOrRedemptionFeesConfs {
+			return false
+		}
```

Comment:
Why wait until 10 confirmations to get a fee update? Is this in case there's a reorg and the fee changes? In that case, it's up to the wallet to determine that a value is not yet safe to report.

Reply:
Will move to wallet. We still want to wait 10 confs though right? Going by this comment https://github.com/decred/dcrdex/pull/1752#discussion_r952515671

Reply:
Personally I think it's a little silly to wait for more than 2 given this is just ancillary info that has no bearing on the success of the swap.  But whatever you guys think...

Reply:
Two sounds good to me.